foursight_core/react/api/encoding_utils.py

```diff
@@ -0,0 +1,42 @@
+import base64
+import json
+
+
+def base64_encode(value) -> str:
+    if not value:
+        return ""
+    if isinstance(value, dict) or isinstance(value, list):
+        value = json.dumps(value)
+    return _base64_encode_to_bytes(value).decode("utf-8")
+
+
+def base64_decode(value: str) -> str:
+    if not value:
+        return ""
+    return _base64_decode_to_bytes(value).decode("utf-8")
+
+
+def _base64_encode_to_bytes(string_or_bytes) -> bytes:
+    if isinstance(string_or_bytes, str):
+        return base64.b64encode(_string_to_bytes(string_or_bytes))
+    elif isinstance(string_or_bytes, bytes):
+        return base64.b64encode(string_or_bytes)
+    else:
+        return bytes("", "utf-8")
```

Comment:
Are we sure we don't want to raise an error for the else clause here? Or if `None` is allowed, then maybe type hint the arg as `Optional[Union[str, bytes]]` or `Union[str, bytes, None]` so that `None` is allowed if you mean to. And then still `-> bytes`. But the real question is if I get something like `4`, should it really quietly return `b''`?

Maybe this happens in some situation where we can't tolerate errors, but then again if we're getting data we don't understand, we should fix it. @willronchetti, thoughts?

Reply:
+1 for raising exception, that way we can be aware when it does happen and handle it correctly later as cases come up